bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/CurrencyProvider.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.library.unit;
+
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.function.Function;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.dimension.Currency;
+
+/**
+ * The {@link CurrencyProvider} is a
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public interface CurrencyProvider {
+
+    Unit<Currency> getSystemCurrency();
```

Comment:
System currency may not always apply as from exchange/conversion point of view you always get pair of currencies. Separating `CurrencyProvider` into i18n/l10n specific and `ExchangeProvider` / `CurrencyConversionProvider` will make it clear.

Reply:
`SystemCurrency` may not be the best naming. It is a sort of "base currency", that is a used for conversion between different currencies. Separating that into an `ExchangeProvider` and `CurrencyProvider` makes it much more difficult.

Reply:
Its easier to introduce complete design now than later. Once this code is part of codebase you will need to look for multiple compromises. Same thing apply to naming.

Reply:
True, but the UoM system requires a base unit for conversions, so we have to set one in any case. UoM also requires that we can convert any unit to any other compatible unit, so if we have "CurrencyA" and "CurrencyB" with an exchange rate "AB" and a "CurrencyC" and "CurrencyC" with an exchange rate "CD" but no exchange rate for "CurrencyC" or "CurrencyD" to "CurrencyA" to "CurrencyB" we have units that we can't convert. Even if we have say "BD" it is very hard to determine the correct exchange rate for "AC", because we would have to check every possible combination if we find one that allows conversion (in this case A->B->D->C). It is far easier (also for the implementation of the exchange rate provider) to have exchanges rates relative to a given "default" (the base unit), in which case at most two conversions (A->base->C) are required.

That said: probably this is already over-engineered, since we are talking about a home automation system where probably no one needs more than one currency.

Reply:
My earlier nitpick was mainly about naming, not functionality. I see you made some adjustments in naming which makes new API is less confusing. I'm happy with currency handling will become part of codebase.
My knowledge of indriya internals is fairly limited, thus I can't comment cleverness of algorithms applied there.